From c5d08c9288890fdfba855d4772240079a7cdc871 Mon Sep 17 00:00:00 2001
From: somebody <no@thankyou.com>
Date: Tue, 25 Mar 2024 22:30:00 +0800
Subject: [PATCH] my modifications

Change default cflags and lan ip;
Enable unlink and tree.
---
 include/target.mk                            | 2 +-
 package/base-files/files/bin/config_generate | 2 +-
 package/utils/busybox/Config-defaults.in     | 4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/target.mk b/include/target.mk
index ac414ef..cc1c0a0 100644
--- a/include/target.mk
+++ b/include/target.mk
@@ -223,7 +223,7 @@ LINUX_RECONF_DIFF = $(SCRIPT_DIR)/kconfig.pl - '>' $(call __linux_confcmd,$(filt
 ifeq ($(DUMP),1)
   BuildTarget=$(BuildTargets/DumpCurrent)
 
-  CPU_CFLAGS = -Os -pipe
+  CPU_CFLAGS = -O2 -pipe
   ifneq ($(findstring mips,$(ARCH)),)
     ifneq ($(findstring mips64,$(ARCH)),)
       CPU_TYPE ?= mips64
diff --git a/package/base-files/files/bin/config_generate b/package/base-files/files/bin/config_generate
index 6589897..c1cd947 100755
--- a/package/base-files/files/bin/config_generate
+++ b/package/base-files/files/bin/config_generate
@@ -162,7 +162,7 @@ generate_network() {
 		static)
 			local ipad
 			case "$1" in
-				lan) ipad=${ipaddr:-"192.168.1.1"} ;;
+				lan) ipad=${ipaddr:-"192.168.6.1"} ;;
 				*) ipad=${ipaddr:-"192.168.$((addr_offset++)).1"} ;;
 			esac
 
diff --git a/package/utils/busybox/Config-defaults.in b/package/utils/busybox/Config-defaults.in
index e5aac86..bebe71a 100644
--- a/package/utils/busybox/Config-defaults.in
+++ b/package/utils/busybox/Config-defaults.in
@@ -898,7 +898,7 @@ config BUSYBOX_DEFAULT_UNIQ
 	default y
 config BUSYBOX_DEFAULT_UNLINK
 	bool
-	default n
+	default y
 config BUSYBOX_DEFAULT_USLEEP
 	bool
 	default n
@@ -2200,7 +2200,7 @@ config BUSYBOX_DEFAULT_TIME
 	default y
 config BUSYBOX_DEFAULT_TREE
 	bool
-	default n
+	default y
 config BUSYBOX_DEFAULT_TS
 	bool
 	default n
